packages/bitcoin/src/utils/bitcoin.descriptors.ts

```diff
@@ -12,3 +18,59 @@ export function getDescriptorFromKeychain<T extends { keyOrigin: string; xpub: s
       return undefined;
   }
 }
+export function deriveAddressFromDescriptor(
+  descriptor: string,
+  network: BitcoinNetworkModes
+): string | undefined {
+  try {
+    const xpubMatch = descriptor.match(/xpub[1-9A-HJ-NP-Za-km-z]{79,108}/);
+    if (!xpubMatch) return undefined;
+
+    const xpub = xpubMatch[0];
+    const accountKeychain = HDKey.fromExtendedKey(xpub);
+    const addressIndexKeychain = deriveAddressIndexZeroFromAccount(accountKeychain);
+
+    if (!addressIndexKeychain.publicKey) return undefined;
+
+    if (descriptor.startsWith('tr(')) {
+      const payment = btc.p2tr(
+        addressIndexKeychain.publicKey,
+        undefined,
+        getBtcSignerLibNetworkConfigByMode(network)
+      );
+      return payment.address;
+    } else if (descriptor.startsWith('wpkh(')) {
+      const payment = btc.p2wpkh(
+        addressIndexKeychain.publicKey,
+        getBtcSignerLibNetworkConfigByMode(network)
+      );
+      return payment.address;
+    }
+
+    return undefined;
+  } catch {
+    return undefined;
+  }
+}
```

Comment:
Can you help me understand why we need this? Is accessing the first native segwit address in the mobile codebase difficult currently?

The problem with this is that, at the package level, we're baking in assumptions that its only the first address that matters. I don't think we should do this, because that isn't how bitcoin works.